Makefile

```diff
@@ -76,43 +76,30 @@ $(BUILDDIR)/:
 ###                                Protobuf                                 ###
 ###############################################################################
 
-PROTO_INPUT_DIR := ./proto
-PROTO_OUTPUT_DIR := ./proto
-PROTO_PLUGIN := gogofaster
-PROTO_PLUGIN_OPTS := Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/golang/protobuf/ptypes/duration,plugins=grpc,paths=source_relative
-PROTO_INCLUDES := -I=$(PROTO_INPUT_DIR) -I=./third_party/proto/
-PROTO_COMPILE := protoc $(PROTO_INCLUDES) --$(PROTO_PLUGIN)_out=$(PROTO_PLUGIN_OPTS):$(PROTO_OUTPUT_DIR)
-
-check-proto-gen-deps:
-ifeq (,$(shell which protoc))
-	$(error "Protocol buffers compiler protoc is required. See https://github.com/protocolbuffers/protobuf/#protocol-compiler-installation for installation instructions.")
+check-proto-deps:
+ifeq (,$(shell which buf))
+	$(error "buf is required for Protobuf building, linting and breakage checking. See https://docs.buf.build/installation for installation instructions.")
```

Comment:
Not, I think, for building?

Reply:
Oh these recent changes demonstrate how to use buf for building in exactly the same way that we use protoc. We don't even need the `third_party` directory any more (buf grabs gogo's protos from the registry now), but I've left it there because #8065 will need it.

You'll see I've left all the `go_package` options in and haven't enabled buf's managed mode. That's the only way I can get buf to build our protos in the current configuration. It's probably also a good idea to leave it like that while #8065 uses that `.intermediate` approach, which is definitely not compatible with buf.

Reply:
#8065 can updated to use this same mechanism for proto generation. The only thing required by the PR is that we can 1. host a copy of the intermediate protos, and 2. build the protos with the `.intermediate` protos copied across. I don't believe it will need anything special for #8065.